zetta_utils/layer/db_layer/datastore/backend.py

```diff
@@ -175,6 +157,67 @@ def write(self, idx: DBIndex, data: DBDataT):
         # must write parent entities for aggregation query to work on `Row` entities
         self.client.put_multi(parent_entities + entities)
 
+    def _get_row_col_keys(self, row_keys: list[str], str_keys: bool = True) -> dict:
+        result = {}
+        for row_key_str in row_keys:
+            row_key = self.client.key("Row", row_key_str)
+            _query = self.client.query(kind="Column", ancestor=row_key)
+            _query.keys_only()
+            if str_keys:
+                result[row_key_str] = tuple(ent.key.id_or_name for ent in _query.fetch())
+            else:
+                result[row_key] = tuple(ent.key for ent in _query.fetch())
+        return result
+
+    @retry(
+        retry=retry_if_not_exception_type((KeyError, RuntimeError, TypeError, ValueError)),
+        stop=stop_after_attempt(3),
+        wait=wait_exponential(multiplier=1, min=4, max=10),
+    )
+    def clear(self, idx: DBIndex | None = None):
+        if idx:
+            keys: list[Key] = []
+            for _key, col_keys in self._get_row_col_keys(idx.row_keys, str_keys=False).items():
+                keys.extend(col_keys)
+                keys.append(_key)
+            self.client.delete_multi(keys=keys)
+            return
```

Comment:
Nag: I think `if/else` is slightly preferable to `if-return` in terms of readability, as it makes the exclusivity of the two options visually explicit